incubator/hnc/internal/pkg/selectors/selectors.go

```diff
@@ -61,23 +61,35 @@ func GetTreeSelector(inst *unstructured.Unstructured) (labels.Selector, error) {
 
 	segs := strings.Split(treeSelectorStr, ",")
 	selectorStr := ""
+	nonNegatedNses := []string{}
 	for i, seg := range segs {
 		// check if it's a valid namespace name
 		if err := validateTreeSelectorSegment(seg); err != nil {
 			return nil, err
 		}
 
+		if strings.TrimSpace(seg)[0] != '!' {
```

Comment:
Why are you trimming the spaces of `seg` here but not, say, on line 75?

Reply:
Because here we need to check if the leading char is `!` so we need to get rid of any leading whitespace. On line 75, `selectorStr` will be passed to `ParseToLabelSelector()`, and from experiments it will handle the whitespaces there.

Reply:
Ahh good to know. But. Hmm. If there were _trailing_ whitespace, though, line 75 wouldn't work.

I feel like it would both be more correct and also easier to read to insert this after line 65:

`seg = strings.TrimSpace(seg)`

And then this line just becomes `if seg[0] != '!'` which is way easier to read.

Reply:
I agree, done!